### PR TITLE
[Live] Handle offline networking elegantly

### PR DESCRIPTION
**Pull request by orta**
Re-uses the view offline structure from our home feed in the Live Auctions, offering people a chance to reconnect, or to bail.

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |253.9ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-handle_bad_networks/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
196.9ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-handle_bad_networks/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
185.3ms | [WebSocket.swift](/artsy/eigen/tree/orta-handle_bad_networks/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift [hunk 1]
```diff
@@ -34,28 +37,68 @@ class LiveAuctionViewController: UISplitViewController {
...
 
-        self.ar_presentIndeterminateLoadingIndicatorAnimated(true)
+        preferredDisplayMode = .AllVisible;
+        preferredPrimaryColumnWidthFraction = 0.4;
+        delegate = self;
 
+        ar_presentIndeterminateLoadingIndicatorAnimated(true)
+        connectToNetwork()
+
+        super.viewWillAppear(animated)
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift]: Is there a reason to put the `super` call at the end here? Could use a comment if so.

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift [hunk 2]
```diff
@@ -34,28 +37,68 @@ class LiveAuctionViewController: UISplitViewController {
...
+                self?.offlineView?.removeFromSuperview()
                 self?.sale = sale
                 self?.setupWithSale(sale, jwt: jwt, bidderID: bidderID)
+
             case .Error(let error):
-                // TODO: handle error case
                 print("Error pulling down sale data for \(self?.saleSlugOrID)")
                 print("Error: \(error)")
-
+                self?.showOfflineView()
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift]: Cool, this works well for now, but after we take a look at auctions-wide error states, we'll have to revisit.

**Comment by ashfurrow** [comment]
💯

**Pull request merged**